machineconfiguration/v1alpha1/types_internalreleaseimage.go

```diff
@@ -0,0 +1,145 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=internalreleaseimages,scope=Cluster
+// +kubebuilder:subresource:status
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/2510
+// +openshift:file-pattern=cvoRunLevel=0000_80,operatorName=machine-config,operatorOrdering=01
+// +openshift:enable:FeatureGate=NoRegistryClusterOperations
+// +kubebuilder:metadata:labels=openshift.io/operator-managed=
+// +kubebuilder:validation:XValidation:rule="self.metadata.name == 'cluster'",message="internalreleaseimage is a singleton, .metadata.name must be 'cluster'"
+
+// InternalReleaseImage is used to keep track and manage a set
+// of release bundles (OCP and OLM operators images) that are stored
+// into the control planes nodes.
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type InternalReleaseImage struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec describes the configuration of this internal release image.
+	// +required
+	Spec InternalReleaseImageSpec `json:"spec,omitzero"`
+
+	// status describes the last observed state of this internal release image.
+	// +optional
+	Status *InternalReleaseImageStatus `json:"status,omitempty,omitzero"`
```

Comment:
So on this one, it's a pointer, which means omitzero isn't needed

However... why is it a pointer? This means you have no required status fields? What does `status: {}` mean?

Reply:
I think the initial `+optional` was to allow the user (installer) to create a first CR without the status - ie, the controller will take care of filling the status accordingly later. The `make lint / verify` goals then insisted on using a pointer, and my understanding was to avoid having a `status: {}` serialized. I can try to review it though

Reply:
Ok, so from a controller point of view I'd expect that the status will never be totally empty: since there will be always at least one spec.releases defined (it cannot be empty), then it's expected that the controller will populate somehow the status at the very first reconciliation (and will keep doing it for every spec.releases entry listed).